the-collab-lab/tcl-45-smart-shopping-list
# Add accessible modal window for item deletion

## Description



This PR replaces the window.confirm() with an accessible dialog modal, which also looks nicer.



Closes #38
Closes #57 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

<img width="554" alt="confirm dialog before the changes" src="https://user-images.githubusercontent.com/63440229/186723409-231c8e4e-f77a-4018-b0f9-a31681461efd.png">

### After

<img width="632" alt="accessible modal to confirm item deletion after the changes" src="https://user-images.githubusercontent.com/63440229/186723561-9edb3537-5670-4fca-a386-6f2e9a6ac70e.png">

## Testing Steps / QA Criteria

1. Checkout branch `NZ-modal-window-for-delete`, run `npm install --save react-a11y-dialog` to install the library and run `npm start`
2. If your list is empty - add an item.
3. Click Delete to initiate the dialog window. Try clicking `X` button to close the modal. Try clicking `Cancel` to cancel the deletion and close the modal. Try clicking `OK` to confirm the deletion - you should see that the modal closes and the item gets deleted from the list.
